sor/src/main/java/com/bazaarvoice/emodb/sor/compactioncontrol/DefaultCompactionControlSource.java

```diff
@@ -86,7 +94,21 @@ public StashRunTimeInfo getStashTime(String id, String dataCenter) {
                 : ImmutableMap.of();
     }
 
-    private String zkKey(String id, String dataCenter) {
+    @VisibleForTesting
+    protected static String zkKey(String id, String dataCenter) {
         return id + "-" + dataCenter;
     }
+
+    @VisibleForTesting
+    protected static Map<String, StashRunTimeInfo> getAllStashTimesWithIdOnlyKeys  (Map<String, StashRunTimeInfo> stashTimesFromZk, List<String> allDatacenterNames) {
+        Map<String, StashRunTimeInfo> allStashTimes = Maps.newHashMap();
+        for (Map.Entry<String, StashRunTimeInfo>  stashTimeFromZk : stashTimesFromZk.entrySet()) {
+            // remove the datacenter string from the key.
+            String stashTimeKey = allDatacenterNames.stream().reduce(stashTimeFromZk.getKey(), (stashTime, dc) -> stashTime.replaceAll(dc, ""));
```

Comment:
This seems risky.  What if the ID is "us-east-1-prod-test" and the resulting key in ZK is "us-east-1-prod-test-us-east-1-prod".  This code would convert the string to "-test-".  May be better to do one of the following:

1. Do a one-time pass to remove suffixes matching "-<datacenter>".  (Don't want to do it twice otherwise a weird but possible ID of "test-us-east-1-prod" in data center eu-west-1 and ZK key "test-us-east-1-prod-eu-west-1-prod" may be stripped down to just "test".)
2. Instead of using "-" use a character which is valid in a ZK string but is not permitted in a data center or compaction control ID as the separator, such as "@".  This way you don't even need to know the data centers, you can just chop everything starting with the separator.  There's a backwards compatibility issue, but so long as no compaction control entries exist at upgrade time it should be ok

Reply:
Yeah, I agree with using @ or something. That way this is way more simpler.